src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -155,6 +155,11 @@ ciMethod::ciMethod(const methodHandle& h_m, ciInstanceKlass* holder) :
     ciReplay::initialize(this);
   }
 #endif
+
+  if (CompilerOracle::should_blackhole(h_m) &&
+      _signature->return_type()->basic_type() == T_VOID) {
```

Comment:
Does it make sense to print a diagnostic message (possibly turned off by `-XX:CompileCommand=quiet`) if a method matches, but has non-void return type?

Reply:
Yes, it does make sense. The check should be moved to `CompilerOracle` to make it cleaner, moved. Added the test a well.